### PR TITLE
Mildm8nnered xct specific matcher improvements

### PR DESCRIPTION
**Pull request by mildm8nnered**
WIP

### CODE REVIEW

File: Source/SwiftLintBuiltInRules/Rules/Idiomatic/XCTSpecificMatcherRuleExamples.swift [hunk 1]
```diff
@@ -63,7 +63,12 @@ internal struct XCTSpecificMatcherRuleExamples {
         // Skip if one operand might be a type or a tuple
         Example("XCTAssert(foo.self == bar)"),
         Example("XCTAssertTrue(type(of: foo) != Int.self)"),
-        Example("XCTAssertTrue(a == (1, 3, 5)")
+        Example("XCTAssertTrue(a == (1, 3, 5)"),
+
+        // Optionals should not trigger
+        Example("XCTAssert(bar == bar.foo?.bar)"),
```
Comment (SimplyDanny) [inline Source/SwiftLintBuiltInRules/Rules/Idiomatic/XCTSpecificMatcherRuleExamples.swift:69]: Try `XCTAssert(bar == bar?.foo.bar)` as well.

File: Source/SwiftLintBuiltInRules/Rules/Idiomatic/XCTSpecificMatcherRuleExamples.swift [hunk 2]
```diff
@@ -110,6 +115,10 @@ internal struct XCTSpecificMatcherRuleExamples {
         Example("↓XCTAssertEqual(foo?.bar, nil)"),
         Example("↓XCTAssertNotEqual(foo?.bar, nil)"),
 
+        // Non-optional chaining and unconditional downcasting
+        Example("↓XCTAssert(bar == bar.foo.bar)"),
+        Example("↓XCTAssert(bar as Foo == foo.bar)"),
```
Comment (SimplyDanny) [inline Source/SwiftLintBuiltInRules/Rules/Idiomatic/XCTSpecificMatcherRuleExamples.swift:120]: Check also that the versions with exclamation marks still trigger.

**Comment by mildm8nnered** [comment]
Hmmm. Actually I think these cases are being handled correctly right now after all

**Comment by SimplyDanny** [comment]
> Hmmm. Actually I think these cases are being handled correctly right now after all

What do you mean?

**Pull request closed without merging**